### ISSUE TITLE
Request failed with 401 Unauthorized

### ISSUE THREAD

**Issue by AbdellM** [RSC-Labs/backstage-changelog-plugin#8]
When attempting to add the Changelog plugin to my Backstage app with the new backend using the legacyPlugin method.
<img width="522" alt="Screenshot 2024-05-04 at 21 54 52" src="https://github.com/RSC-Labs/backstage-changelog-plugin/assets/71790032/4223f829-e043-4260-bfe3-df840f99cc23">

I encountered a 401 Unauthorized error, even though the Bearer token was sent from the frontend. 
![Screenshot 2024-05-04 at 21 56 49](https://github.com/RSC-Labs/backstage-changelog-plugin/assets/71790032/f05771d8-d924-49f2-ba8d-e7174a218f12)

The error message states "Illegal token".
![Screenshot 2024-05-04 at 21 51 03](https://github.com/RSC-Labs/backstage-changelog-plugin/assets/71790032/e6a53700-1d41-43bb-87bc-d7069ac102ab)

**Comment by radoslaw-sz** [comment]
hi @AbdellM  - I am sorry to hear that, but unfortunately this plugin has been developed with the old backend system and was never tested against new one.
Due to other, high priority activites I cannot focus on that too much now, but did you maybe check other plugins just for testing if they are properly working? It looks like general problem, not related to this plugin, but of course - I can be wrong.

**Comment by AbdellM** [comment]
After debugging, I found out that the token returned in this line wasn't the same Bearer token that was sent from the frontend. This is why I got an error message "Illegal token".
https://github.com/RSC-Labs/backstage-changelog-plugin/blob/5c484a74a7094b2a1400b132b5cd732326223278/plugins/backstage-changelog-plugin-backend/src/service/router.ts#L54

There is the fix that made it work for me; I extracted the token from the request
> ```diff
>+  const token = req.header("Authorization").split(' ')[1];
>    const { namespace, kind, name } = req.params;
>    const entity = await catalogApi.getEntityByRef(
>      { namespace, kind, name },
>+    { token: token }
>    );
> ```
However, I don't know if that affects something else.
